R/output-ipc-functions.R

```diff
@@ -1 +1,64 @@
-# TODO: add write_ipc and sink_ipc
+# TODO: add write_ipc
+
+#' Evaluate the query in streaming mode and write to an IPC file
+#'
+#' @inherit lazyframe__sink_parquet description params return
+#' @inheritParams rlang::args_dots_empty
+#' @param compression Must be one of:
+#' * `"uncompressed"`;
+#' * `"lz4"`: fast compression/decompression;
+#' * `"zstd"`: good compression performance.
+#'
+#' If `NULL`, it uses `"uncompressed"`.
+#'
+#' @examples
+#' tmpf <- tempfile()
+#' as_polars_lf(mtcars)$sink_ipc(tmpf)
+#' pl$scan_ipc(tmpf)$collect()
+lazyframe__sink_ipc <- function(
+  path,
+  ...,
+  compression = c("zstd", "lz4", "uncompressed"),
+  maintain_order = TRUE,
+  type_coercion = TRUE,
+  `_type_check` = TRUE,
+  predicate_pushdown = TRUE,
+  projection_pushdown = TRUE,
+  simplify_expression = TRUE,
+  slice_pushdown = TRUE,
+  collapse_joins = TRUE,
+  no_optimization = FALSE,
+  storage_options = NULL,
+  retries = 2
+) {
+  wrap({
+    check_dots_empty0(...)
+    compression <- compression %||% "uncompressed"
+    compression <- arg_match0(compression, values = c("zstd", "lz4", "uncompressed"))
+    if (compression == "uncompressed") {
+      compression <- NULL
+    }
```

Comment:
Convert `NULL` to `"uncompressed"` and then back to `NULL` so that `arg_match()` doesn't fail when it receives `NULL`.